src/platform/silabs/efr32/BLEManagerImpl.cpp

```diff
@@ -949,6 +949,7 @@ CHIP_ERROR BLEManagerImpl::EncodeAdditionalDataTlv()
 
     err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(additionalDataPayloadParams, c3AdditionalDataBufferHandle,
                                                                          additionalDataFields);
+    SuccessOrExit(err);
```

Comment:
why are we adding this here? we are already at the exit
```suggestion
```

Reply:
I'd recommend changing the function to remove the exit/label approach

Reply:
Removed exit label and added `VerifyOrReturnError`.